src/model.ts

```diff
@@ -29,14 +44,21 @@ export type MarkdownNil = false | null | undefined;
 /** Any primitive Markdown type. */
 export type MarkdownNode = MarkdownNil | MarkdownText | MarkdownElement;
 
+/** Promise Markdown type that resolves to nested Markdown type. */
+export type MarkdownPromise = Promise<MarkdownChildren>;
+
 /** Nested Markdown type. */
-export type MarkdownChildren = MarkdownNode | MarkdownChildren[];
+export type MarkdownChildren =
+  | MarkdownNode
+  | MarkdownChildren[]
+  | MarkdownPromise;
```

Comment:
MarkdownPromise should not be needed here, as it already is part of MarkdownNode via MarkdownElement